src/client/components/BaselayerControl.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <button
+    class="baselayer-control"
+    @click="switchBaseLayer"
+  >
+    <span class="a11y-sr-only">Switch to {{ nextLayer.title }}</span>
+    <img
+      :src="nextLayer.thumbnail"
+      :alt="nextLayer.title"
+      class="baselayer-control__image"
+    >
+  </button>
+</template>
+
+<script>
+export default {
+  props: {
+    layers: {
+      type: Array,
+      required: true,
+      validator (layers) {
+        return layers.reduce((isValid, layer) => {
+          return isValid
+            && typeof layer.title === 'string'
+            && typeof layer.thumbnail === 'string'
+            && typeof layer.style === 'string'
+        }, true)
+      }
+    },
+    switchHandler: {
```

Comment:
Why is it a prop and not handled by an emitted event?

Reply:
Because this component is rendered on the mapbox map. So it doesn't interact with the Vue environment directly